operator/v1/types.go

```diff
@@ -175,6 +175,16 @@ const (
 	ConditionUnknown ConditionStatus = "Unknown"
 )
 
+// StaticPodOperatorSpec is spec for controllers that manage static pods.
+type StaticPodOperatorSpec struct {
+	OperatorSpec           `json:",inline"`
+
+	// failedRevisionLimit is the number of failed static pod installer revisions to keep on disk and in the api, with 0 being unlimited
+	FailedRevisionLimit    int `json:"failedRevisionLimit"`
```

Comment:
`FailedRevisionHistoryLimit`  similar down below.

We don't use ints, use int32

zero is not unlimited.  zero is the default.  -1 can be unlimited

omitempty